scripts/getAnchorPrices.ts

```diff
@@ -0,0 +1,252 @@
+        id: feedDataWithProof.body.id,
+        value: feedDataWithProof.body.value,
+        turnoutBIPS: feedDataWithProof.body.turnoutBIPS,
+        decimals: feedDataWithProof.body.decimals,
+      },
+    });
+
+    return isValid;
+  } catch (error) {
+    const feedId = feedDataWithProof?.body?.id || "unknown";
+    console.error(`Error verifying feed data for ID ${feedId}:`, error);
+    return false;
+  }
+}
+
+async function fetchPriceData(apiUrl: string, feedId: string): Promise<any> {
+  const response = await fetch(apiUrl, {
+    method: "POST",
+    headers: {
+      "x-apikey": process.env.FLARE_API_KEY || "",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      feed_ids: [feedId],
+    }),
+  });
+
+  if (!response.ok) {
+    const errorBody = await response.text();
+    throw new Error(
+      `HTTP error! Status: ${response.status} for feed_id: ${feedId}. Body: ${errorBody}`
+    );
+  }
+
+  const responseData = await response.json();
+  if (!Array.isArray(responseData) || responseData.length === 0) {
+    throw new Error(
+      `Unexpected response format or empty data for feed_id: ${feedId}`
+    );
+  }
+  return responseData[0];
+}
+
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+async function processPriceFeeds(
+  ftsoV2Contract: Contract | any,
+  apiUrlGetProofs: string,
+  feedIds: string[]
+): Promise<PriceDataResult> {
+  const allResponses = [];
+  const verificationResults = [];
+
+  for (const feedId of feedIds) {
+    try {
+      const feedData = await fetchPriceData(apiUrlGetProofs, feedId);
+      allResponses.push(feedData);
+
+      const isValid = await verifyFeedDataOnChain(ftsoV2Contract, feedData);
+      console.log(`Verification for ${feedId}: ${isValid}`);
+      verificationResults.push({
+        feedId,
+        isValid,
+        data: feedData,
+      });
+
+      await delay(200);
+    } catch (error: any) {
+      console.error(`Failed to process feed ${feedId}:`, error.message);
+      verificationResults.push({
+        feedId,
+        isValid: false,
+        data: null,
+        error: error.message,
+      });
+      await delay(100);
+    }
+  }
+
+  return {
+    responses: allResponses,
+    verifications: verificationResults,
+  };
+}
+
+async function main() {
+  try {
+    const networkName = hre.network.name;
+
+    const currentNetworkNamespace = networkNamespaces[networkName];
+    if (!currentNetworkNamespace) {
+      throw new Error(
+        `Unsupported network: ${networkName}. Must be one of: ${Object.keys(networkNamespaces).join(", ")}`
+      );
+    }
+
+    const currentNetwork = networks[networkName];
+    if (!currentNetwork || !currentNetwork.url) {
+      throw new Error(
+        `Network configuration or DA Layer URL not found for ${networkName} in .env file`
+      );
+    }
+
+    const apiUrlFeedNames = `${currentNetwork.url}/api/v0/ftso/anchor-feed-names`;
```

Comment:
Let's make a global utils to fetch the correct url depending on the network - to remove the if over env file COSTON/COSTON2